src/theory/strings/theory_strings_rewriter.cpp

```diff
@@ -1058,17 +1094,21 @@ RewriteResponse TheoryStringsRewriter::preRewrite(TNode node) {
           vec2.push_back(n);
           for(unsigned j=l; j<u; j++) {
             vec_nodes.push_back(r);
-            n = vec_nodes.size()==1? r : prerewriteConcatRegExp(NodeManager::currentNM()->mkNode(kind::REGEXP_CONCAT, vec_nodes));
+            n = vec_nodes.size() == 1 ? r
```

Comment:
Suggestion: it might be a good idea to introduce a `mkRegexpConcat` helper function that does the `size == 1` case since this pattern is so common.

Reply:
Already have it actually: `mkConcat`, I'll use.